### PR TITLE
Query data from multiple tables using sql joins

### PR DESCRIPTION
**Pull request by NoelLincoln**
Hello 👋,

In this milestone I was able to create these tables:
- 'vets'
- 'specializations' table to handle the relationship between species and vets.
- 'visits' to handle this between animals and vets.

I added the data as per the data.sql file

I finally queried data from the tables :

- Who was the last animal seen by William Tatcher?
last animal seen by William Tatcher

- How many different animals did Stephanie Mendez see?
How many different animals did Stephanie Mendez see

- List all vets and their specialties, including vets with no specialties.
all vets and their specialties, including vets with no specialties

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
all animals that visited Stephanie Mendez between April 1st and August 30th, 2020

- What animal has the most visits to vets?
What animal has the most visits to vets

- Who was Maisy Smith's first visit?
Maisy Smith's first visit

- Details for most recent visit: animal information, vet information, and date of visit.
Details for most recent visit animal information vet information and date of visit

- How many visits were with a vet that did not specialize in that animal's species?
How many visits were with a vet that did not specialize in that animal's species

- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
What specialty should Maisy Smith consider getting the species she gets the most

Cheers!

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -228,3 +228,144 @@ GROUP BY
...
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT
+    a.name AS animal_name
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+WHERE
+    v.vet_id = (SELECT id FROM vets WHERE name = 'Stephanie Mendez')
+    AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (franclobo) [inline queries.sql:280]: - There are two visits to Stephanie Mendez from April 1st and August 30th, 2020. Those are 👇🏽 

> Agumon visited Stephanie Mendez on Jul 22th, 2020.
> Blossom visited Stephanie Mendez on May 24th, 2020.

Please 🙏🏽 fix this query to have the correct information.

File: queries.sql [hunk 2]
```diff
@@ -228,3 +228,144 @@ GROUP BY
...
+    COUNT(v.animal_id) AS visit_count
+FROM
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+GROUP BY
+    a.name
+ORDER BY
+    visit_count DESC
+LIMIT 1;
```
Comment (franclobo) [inline queries.sql:296]: - [ ] Boarmon has the most visits to the vet these are 4, please 🙏🏽 fix this query to have the correct information.

> Boarmon visited Maisy Smith on Jan 24th, 2019.
> Boarmon visited Maisy Smith on May 15th, 2019.
> Boarmon visited Maisy Smith on Feb 27th, 2020.
> Boarmon visited Maisy Smith on Aug 3rd, 2020.

File: queries.sql [hunk 3]
```diff
@@ -228,3 +228,144 @@ GROUP BY
...
+
+SELECT
+    s.name AS suggested_specialty
+FROM
+    VetSpeciesVisits vsv
+JOIN
+    species s ON vsv.species_id = s.id
+ORDER BY
+    vsv.num_visits DESC
+LIMIT 1;
```
Comment (franclobo) [inline queries.sql:371]: - [ ] Maisy Smith has 9 visits, 6 Digimon, and 3 Pokemon. Please 🙏🏽 fix this query to have the correct information.

File: queries.sql [hunk 4]
```diff
@@ -228,3 +228,144 @@ GROUP BY
...
+    visits v
+JOIN
+    animals a ON v.animal_id = a.id
+JOIN
+    vets ve ON v.vet_id = ve.id
+LEFT JOIN
+    specializations s ON ve.id = s.vet_id AND a.species_id = s.species_id
+WHERE
+    s.vet_id IS NULL
+    OR s.species_id IS NULL;
```
Comment (franclobo) [inline queries.sql:342]: - Maisy Smith doesn't have specialization so the number of visits should be more than 9 visits. Please 🙏🏽 fix this query to have the correct information.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @NoelLincoln, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @NoelLincoln, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. You are almost there...



## Highlights

- Added queries, schema and data in appropriate files ✅ 
- Descriptive pull request with screenshots ✅ 

 ## Required Changes ♻️

- [ ] I have noticed that you have not made any changes in your currently submitted PR. Please make sure that you address all the changes requested by the previous reviewer before submitting another request.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @NoelLincoln, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Worked on previous reviewer comments
✅ Updated screenshots. 👍 
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**